File: IMSE_DBFilling/src/DatabaseHelper.java

import java.sql.*;
import java.time.LocalDate;
import java.util.ArrayList;


class DatabaseHelper {
    private static final String DB_CONNECTION_URL = "jdbc:mariadb://localhost:3306/db";
    private static final String USER = "user";
    private static final String PASS = "user";

    private static Statement stmt;
    private static Connection con;

    private int cos_count = 0;

    
    //CREATE CONNECTION
    DatabaseHelper() {
        try {
            con = DriverManager.getConnection(DB_CONNECTION_URL, USER, PASS);
            stmt = con.createStatement();
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }

    //INSERT INTO
    boolean insertIntoCostumer(String email, int phone_number, String firstname, String surname, int locId ) {

        try {
            String sql = "INSERT INTO COSTUMER (email, phone_number, firstname, surname, LOCATIONID) VALUES ('" +
            		 email +
                     "', '" +
                     phone_number +
                     "', '" +
                     firstname +
                     "', '" +
                    surname +
                    "', '" +
                    locId +
                    "')";
            stmt.execute(sql);
            cos_count++;
        } catch (Exception e) {
            return false;
        }
        return true;
    }

    void insertIntoLocation(int LocId, int zipcode, String street, String city) {
        try {
            String sql = "INSERT INTO LOCATIONS (LOCATION_ID, ZIP_CODE, STREET, CITY) VALUES ('" +
            		LocId +
                    "', '" +
                    zipcode +
                    "', '" +
                    street +
                    "', '" +
                    city +
                    "')";
            stmt.execute(sql);

        } catch (Exception e) {
            System.err.println("Error at: insertIntoLocations\nmessage: " + e.getMessage());
            cos_count--;
        }
    }

    void insertIntoCar(int regnr, String cmodel, int year, int price) {
        try {
            String sql = "INSERT INTO CAR (REGISTRATION_NUMBER, CAR_MODEL, MODEL_YEAR, DAILY_PRICE) VALUES ('" +
            		regnr +
                    "', '" +
                    cmodel +
                    "', '" +
                    year +
                    "', '" +
                    price +
                    "')";
            stmt.execute(sql);
        } catch (Exception e) {
            System.err.println("Error at: insertIntoCar\nmessage: " + e.getMessage());
        }
    }

    void insertIntoBilling(int nr, int total,  String date, String c_mail) {
        try {
            String sql = "INSERT INTO BILLING (BILL_NUMBER, TOTAL_PRICE, BILLDATE, C_EMAIL) VALUES ('" +
            		nr +
                    "', '" +
                    total +
                    "', '" +
                    date +
                    "', '" +
                    c_mail +
                    "')";
            stmt.execute(sql);
        } catch (Exception e) {
            System.err.println("Error at: insertIntoBilling\nmessage: " + e.getMessage());
        }
    }

    void insertIntoReservation(int nr,  String from, String returnd, int amount, int billnr) {
        try {
            String sql = "INSERT INTO RESERVATION (RESERVATION_NUMBER, FROM_DATE, RETURN_DATE, AMOUNT, RESERVATION_BILL_NUMBER) VALUES ('" +
            		nr +
                    "', '" +
                    from +
                    "', '" +
                    returnd +
                    "', '" +
                    amount +
                    "', '" +
                    billnr +
                    "')";
            stmt.execute(sql);
        } catch (Exception e) {
            System.err.println("Error at: insertIntoReservation\nmessage: " + e.getMessage());
        }
    }

    void insertIntoInsurance( int nr, String name) {
        try {
            String sql = "INSERT INTO INSURANCE (INSURANCE_NUMBER, NAME) VALUES ('" +
            	nr+
            	"', '" +
                    name +
                    "')";
            stmt.execute(sql);
        } catch (Exception e) {
            System.err.println("Error at: insertIntoInsurance\nmessage: " + e.getMessage());
        }
    }

    void insertIntoAreFriends( String email1, String email2) {
        try {
            String sql = "INSERT INTO ARE_FRIENDS (FRIEND_COSTUMER_EMAIL_1, FRIEND_COSTUMER_EMAIL_2) VALUES ('" +
                    email1+
                    "', '" +
                    email2 +
                    "')";
            stmt.execute(sql);
        } catch (Exception e) {
            System.err.println("Error at: insertIntoAreFriends\nmessage: " + e.getMessage());
        }
    }

    void insertIntoRent(String email1, int car, int reservation) {
        try {
            String sql = "INSERT INTO RENT(RENT_EMAIL, RENT_CAR, RENT_RESERVATION) VALUES ('" +
                    email1+
                    "', '" +
                    car +
                    "', '" +
                    reservation +
                    "')";
            stmt.execute(sql);
        } catch (Exception e) {
            System.err.println("Error at: insertIntoRent\nmessage: " + e.getMessage());
        }
    }

    //SELECT
    ArrayList<String> selectCostumerEmail() {
        ArrayList<String> emails = new ArrayList<>();

        try {
            ResultSet rs = stmt.executeQuery("SELECT EMAIL FROM COSTUMER ORDER BY EMAIL");
            while (rs.next()) {
                emails.add(rs.getString("EMAIL"));
            }
            rs.close();
        } catch (Exception e) {
            System.err.println(("Error at: selectCostumerEmail\n message: " + e.getMessage()).trim());
        }
        return emails;
    }

    ArrayList<String> selectRENTSize() {
        ArrayList<String> rents = new ArrayList<>();

        try {
            ResultSet rs = stmt.executeQuery("SELECT RENT_EMAIL FROM RENT");
            while (rs.next()) {
                rents.add(rs.getString("RENT_EMAIL"));
            }
            rs.close();
        } catch (Exception e) {
            System.err.println(("Error at: selectRENTSize\n message: " + e.getMessage()).trim());
        }
        return rents;
    }

    ArrayList<Integer> selectRegistrationNumber() {
        ArrayList<Integer> nr = new ArrayList<>();

        try {
            ResultSet rs = stmt.executeQuery("SELECT REGISTRATION_NUMBER FROM CAR");
            while (rs.next()) {
                nr.add(rs.getInt("REGISTRATION_NUMBER"));
            }
            rs.close();
        } catch (Exception e) {
            System.err.println(("Error at: selectRegistrationNumber\n message: " + e.getMessage()).trim());
        }
        return nr;
    }

    ArrayList<Integer> selectLocationId() {
        ArrayList<Integer> IDs = new ArrayList<>();

        try {
            ResultSet rs = stmt.executeQuery("SELECT LOCATION_ID FROM LOCATIONS");
            while (rs.next()) {
                IDs.add(rs.getInt("LOCATION_ID"));
            }
            rs.close();
        } catch (Exception e) {
            System.err.println(("Error at: selectLocationId\n message: " + e.getMessage()).trim());
        }
        return IDs;
    }

    int selectCountAllFromBilling(){
        int count = 0;
        try {
            ResultSet rs = stmt.executeQuery("SELECT COUNT(*) FROM BILLING");
            while (rs.next()) {
                count = rs.getInt(1);
            }
            rs.close();
        } catch (Exception e) {
            System.err.println(("Error at: selectCountAllFromBilling\n message: " + e.getMessage()).trim());
        }
        return count;
    }

    public void close()  {
        try {
            stmt.close();
            con.close();
        } catch (Exception ignored) {
        }
    }
}

File: web/DatabaseHelper.php
<?php

class DatabaseHelper
{
    public function connect(){

        $conn = mysqli_connect("mariadb", "user", "user", "db");

        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        } else {

        }
        return $conn;
    }

    public function selectAllCostumers($conn, $email, $phone, $fname, $lname, $locid)
    {
        $sql = "SELECT * FROM COSTUMER
            WHERE EMAIL LIKE '%{$email}%'
            and PHONE_NUMBER LIKE '%{$phone}'
            and FIRSTNAME LIKE '%{$fname}'
            and SURNAME LIKE '%{$lname}'
            and LOCATIONID LIKE '%{$locid}'
            ORDER BY surname ASC";

        $result = mysqli_query($conn, $sql);

        return $result;
    }


    public function selectAllBill($conn,$billnr, $total, $date, $c_email)
    {
        $sql = "SELECT * FROM BILLING
            WHERE BILL_NUMBER LIKE '%{$billnr}%'
            and TOTAL_PRICE LIKE '%{$total}'
            and BILLDATE LIKE '%{$date}'
            and C_EMAIL LIKE '%{$c_email}'
            ORDER BY BILL_NUMBER ASC";

        $result = mysqli_query($conn, $sql);

        return $result;
    }

    public function selectAllLocations($conn,$locid, $zip_code, $street, $city)
    {
        $sql = "SELECT * FROM BILLING
            WHERE LOCATION_ID LIKE '%{$locid}%'
            and ZIP_CODE LIKE '%{$zip_code}'
            and STREET LIKE '%{$street}'
            and CITY LIKE '%{$city}'
            ORDER BY LOCATION_ID ASC";

        $result = mysqli_query($conn, $sql);

        return $result;
    }

    public function selectAllCars($conn,$reg_num, $model, $price)
    {
        $sql = "SELECT * FROM CAR
            WHERE REGISTRATION_NUMBER LIKE '%{$reg_num}%'
            and CAR_MODEL LIKE '%{$model}'
            and DAILY_PRICE LIKE '%{$price}'
            ORDER BY REGISTRATION_NUMBER ASC";

        $result = mysqli_query($conn, $sql);

        return $result;
    }

    public function selectAllReservations($conn,$resnr, $from, $return, $amount, $bill)
    {
        $sql = "SELECT * FROM RESERVATION
            WHERE RESERVATION_NUMBER LIKE '%{$resnr}%'
            and FROM_DATE LIKE '%{$from}'
            and RETURN_DATE LIKE '%{$return}'
            and AMOUNT LIKE '%{$amount}'
            and RESERVATION_BILL_NUMBER LIKE '%{$bill}'
            ORDER BY RESERVATION_NUMBER ASC";

        $result = mysqli_query($conn, $sql);

        return $result;
    }


    public function selectAllRents($conn, $email, $car, $res)
    {
        $sql = "SELECT * FROM RENT";

        $result = mysqli_query($conn, $sql);

        return $result;
    }


   public function insertIntoCostumer($conn, $email, $phone, $name, $surname, $location)
   {
       $sql = "INSERT INTO Costumer (EMAIL, PHONE_NUMBER, FIRSTNAME, SURNAME, LOCATIONID) VALUES ('{$email}', '{$phone}', '{$name}','{$surname}', '{$location}')";

       if (mysqli_query($conn, $sql)) {
           echo "Insert successfully!";
       } else {
           echo "Error: " . $sql . "<br>" . mysqli_error($conn);
       }
   }


    public function insertIntoBilling($conn, $total, $date, $c_mail)
    {
        $sql = "INSERT INTO BILLING (TOTAL_PRICE, BILLDATE, C_EMAIL) VALUES ('{$total}', '{$date}','{$c_mail}')";

        if (mysqli_query($conn, $sql)) {
            echo "Insert successfully!";
        } else {
            echo "Error: " . $sql . "<br>" . mysqli_error($conn);
        }
    }

    public function insertIntoReservation($conn, $RESERVATION_NUMBER, $FROM_DATE, $RETURN_DATE, $AMOUNT)
    {
        $sql = "INSERT INTO RESERVATION (reservation_number, from_date, return_date, amount, reservation_bill_number) VALUES ('{$RESERVATION_NUMBER}', '{$FROM_DATE}','{$RETURN_DATE}','{$AMOUNT}','{$RESERVATION_NUMBER}')";

        if (mysqli_query($conn, $sql)) {
            echo "Insert successfully!";
        } else {
            echo "Error: " . $sql . "<br>" . mysqli_error($conn);
        }

    }

    public function insertIntoCAR($conn, $REGISTRATION_NUMBER, $CAR_MODEL, $MODEL_YEAR, $DAILY_PRICE)
    {
        $sql = "INSERT INTO CAR (registration_number, car_model, model_year, daily_price) VALUES ('{$REGISTRATION_NUMBER}', '{$CAR_MODEL}','{$MODEL_YEAR}','{$DAILY_PRICE}')";

        if (mysqli_query($conn, $sql)) {
            echo "Insert successfully!";
        } else {
            echo "Error: " . $sql . "<br>" . mysqli_error($conn);
        }
    }

    public function insertIntoRent($conn, $RENT_EMAIL, $RENT_CAR, $RENT_RESERVATION)
    {
        $sql = "INSERT INTO RENT (rent_email, rent_car, rent_reservation) VALUES ('{$RENT_EMAIL}', '{$RENT_CAR}','{$RENT_RESERVATION}')";

        if ($conn->query($sql) === TRUE) {
            echo "Insert successfully!";
        } else {
            echo "Error: " . $sql . "<br>" . mysqli_error($conn);
        }

    }


}